roachtest: allow running roachtests on AWS

petermattis opened the pull request:
Add a `--cloud` flag (default "gce") which allows running roachtests on
"aws". Refactor some of the GCE specific code to also support AWS.

Fixes #29334

Release note: None

petermattis committed:
roachtest: allow running roachtests on AWS

Add a `--cloud` flag (default "gce") which allows running roachtests on
"aws". Refactor some of the GCE specific code to also support AWS.

Fixes #29334

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/31314)
<!-- Reviewable:end -->

petermattis commented:
This is still a WIP as `roachprod` fails when trying to create a cluster on `i3` machines as it is using the Google device names. We'll also need to add another teamcity config for running some subset of the roachtests on AWS.

petermattis commented:
Oh, I think `roachprod` is ok. The problem is `RemountNoBarrier`.

petermattis commented:
Fixed up `RemountNoBarrier` and switched to specifying `--aws-machine-type-ssd` instead of `--aws-machine-type`. Everything seems to be working in simple testing. For example, `roachtest run --cloud=aws kv95/encrypt=false/nodes=1`.

andreimatei reviewed (approved):
LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31314#-:-LOdbWof2WTLnA6FfHe_:bb6ywmx)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cmd/roachtest/cluster.go, line 393 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/31314#-LOdb9t-AhV8ksBAdLr2:-LOdb9t-AhV8ksBAdLr3:b-yhlb71) ([raw file](https://github.com/cockroachdb/cockroach/blob/60c390809c8afe29b4e5edc95151dc562fbed0bb/pkg/cmd/roachtest/cluster.go#L393)):*
> ```Go
> 
> 	fmt.Fprintf(os.Stderr, "unknown machine type: %s\n", s)
> 	os.Exit(1)
> ```

here and elsewhere: `log.Fatalf()`? Is there a reason there's a lot of `os.Exit(1)` in roachtest?
___
*[pkg/cmd/roachtest/main.go, line 152 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/31314#-LOdafGe0_-vJ_7nFMJ4:-LOdafGe0_-vJ_7nFMJ5:b-qsut7u) ([raw file](https://github.com/cockroachdb/cockroach/blob/60c390809c8afe29b4e5edc95151dc562fbed0bb/pkg/cmd/roachtest/main.go#L152)):*
> ```Go
> 			&artifacts, "artifacts", "artifacts", "path to artifacts directory")
> 		cmd.Flags().StringVar(
> 			&cloud, "cloud", cloud, "cloud provider to use (local aws gce)")
> ```

what's "local"? How does that, and this flag in general, interact with "--local"? Should that flag go away?


<!-- Sent from Reviewable.io -->

petermattis reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31314)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cmd/roachtest/cluster.go, line 393 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/31314#-LOdb9t-AhV8ksBAdLr2:-LOdg0ImBCE4-zWOEM0E:bun09p0) ([raw file](https://github.com/cockroachdb/cockroach/blob/60c390809c8afe29b4e5edc95151dc562fbed0bb/pkg/cmd/roachtest/cluster.go#L393)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

here and elsewhere: `log.Fatalf()`? Is there a reason there's a lot of `os.Exit(1)` in roachtest?
</blockquote></details>

I can't recall offhand. I used `fmt.Fprintf` and `os.Exit` for consistency with other roachtest code. I'd like to leave this as-is for this PR. Feel free to determine if there is a good reason for this (i.e. spelunk `git blame`) and switch to using `log.Fatal` if there isn't. Note that using `util/log` isn't a good idea for command line tools as we don't want a stacktrace.
___
*[pkg/cmd/roachtest/main.go, line 152 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/31314#-LOdafGe0_-vJ_7nFMJ4:-LOdgRKW1ztR3UuLQPIp:b-7sd4rs) ([raw file](https://github.com/cockroachdb/cockroach/blob/60c390809c8afe29b4e5edc95151dc562fbed0bb/pkg/cmd/roachtest/main.go#L152)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

what's "local"? How does that, and this flag in general, interact with "--local"? Should that flag go away?
</blockquote></details>

I stole part of the flag description from `roachprod`. Specifying `--cloud=local` would not have worked. Perhaps we can make that work, though I think I'd want to keep `--local` as a synonym for `--cloud=local`. You're doing a bunch of work in this area right now and if I muck with it concurrently we'll likely just have merge conflicts. Feel free to sanitize this in the rest of your efforts, or leave it and I'll fix it up once you're done.

I've removed `local` from the flag description.


<!-- Sent from Reviewable.io -->

petermattis commented:
bors r=andreimatei
